### PR TITLE
FEATURE: Rename Reset Read bulk action to Defer

### PR DESCRIPTION
**Pull request by nbianca**
It is enabled only if defer is enabled in user options too and if the
button shows up in the topic's footer.

### CODE REVIEW

**Review by eviltrout — APPROVED** [review]

File: app/assets/javascripts/discourse/app/templates/components/search-result-entry.hbs [hunk 1]
```diff
@@ -10,7 +10,7 @@
       {{track-selected selectedList=selected selectedId=post.topic class="bulk-select"}}
     {{/if}}
 
-    <a href={{post.url}} {{action "logClick" post.topic_id}} class="search-link" role="heading" aria-level="2">
+    <a href={{post.url}} {{action "logClick" post.topic_id}} class="search-link{{if post.topic.visited " visited"}}" role="heading" aria-level="2">
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/app/templates/components/search-result-entry.hbs:13]: What is this change doing?

File: app/models/post_timing.rb [hunk 2]
```diff
@@ -58,8 +58,8 @@ def self.record_timing(args)
     record_new_timing(args) if rows == 0
   end
 
-  def self.destroy_last_for(user, topic_id)
-    topic = Topic.find(topic_id)
+  def self.destroy_last_for(user, topic_id, topic: nil)
```
Comment (ZogStriP) [inline app/models/post_timing.rb:61]: Not a super fan of the signature of this method. Maybe there should be two methods instead? One accepting a topic_id, another one accepting a topic object. And the one accepting the topic_id would find the topic and then call the other method?

File: app/assets/javascripts/discourse/app/templates/components/search-result-entry.hbs (hunk 1 above)
Comment (nbianca) [reply app/assets/javascripts/discourse/app/templates/components/search-result-entry.hbs]: There was another bug in the report. This was it and it was kind of related to these improvements. I moved it to a different PR: https://github.com/discourse/discourse/pull/15994

File: app/models/post_timing.rb (hunk 2 above)
Comment (nbianca) [reply app/models/post_timing.rb]: I wasn't either, but I did not want to have another method for this. We already have two: destroy_for and destroy_last_for. I changed this a bit, is it better now? If it's not much better, I will write another method.

Comment (ZogStriP) [reply app/models/post_timing.rb]: I guess that works too 👍

File: app/assets/javascripts/discourse/app/templates/components/search-result-entry.hbs (hunk 1 above)
Comment (ZogStriP) [reply app/assets/javascripts/discourse/app/templates/components/search-result-entry.hbs]: I see. Yeah best to extract that in its own PR/commit 👍

**Review by ZogStriP — APPROVED** [review]

**Pull request merged**